src/openshift/component.ts

```diff
@@ -145,6 +144,67 @@ export class Component extends OpenShiftItem {
+            {
+                label: 'Component',
+                description: 'Unlink Component'
+            },
+            {
+                label: 'Service',
+                description: 'Unlink Service'
+            }
+        ];
+        const unlinkActionSelected = await window.showQuickPick(unlinkActions, {placeHolder: 'Select the option'});
+        if (!unlinkActionSelected) return null;
+        return unlinkActionSelected.label === 'Component' ? Component.unlinkComponent(context) : unlinkActionSelected.label === 'Service' ?  Component.unlinkService(context) : null;
+    }
+
+    static async unlinkComponent(context: OpenShiftObject) {
+        const linkCompName: Array<string> = [];
+        const component = await Component.getOpenShiftCmdData(context,
+            'Select a Project',
+            'Select an Application',
+            'Select a Component');
+        if (!component) return null;
+        const linkComponent = await Component.getLinkData(component);
+        const getLinkComponent = linkComponent['status'].linkedComponents;
+        if (!getLinkComponent) throw Error('No linked Components found');
+        Object.keys(getLinkComponent).forEach(async key => {
+            linkCompName.push(key);
+        });
+        const compName = await window.showQuickPick(linkCompName, {placeHolder: "Select a Component to unlink"});
+        if (!compName) return null;
+        return Progress.execFunctionWithProgress(`Unlinking Component`,
+            () => Component.odo.execute(Command.unlinkComponents(component.getParent().getParent().getName(), component.getParent().getName(), component.getName(), compName), component.contextPath.fsPath)
+                .then(() => `Component ${compName} has been successfully unlinked from the component ${component.getName()}`)
+                .catch((err) => Promise.reject(`Failed to unlink component with error '${err}'`))
+        );
+    }
+
+    static async unlinkService(context: OpenShiftObject) {
+        const component = await Component.getOpenShiftCmdData(context,
+            'Select a Project',
+            'Select an Application',
+            'Select a Component');
+        if (!component) return null;
+        const linkService = await Component.getLinkData(component);
+        const getLinkService = linkService['status'].linkedServices;
+        if (!getLinkService) throw Error('No linked Services found');
+        const serviceName = await window.showQuickPick(getLinkService, {placeHolder: "Select a Service to unlink"});
+        if (!serviceName) return null;
+        return Progress.execFunctionWithProgress(`Unlinking Service`,
+            () => Component.odo.execute(Command.unlinkService(component.getParent().getParent().getName(), component.getParent().getName(), serviceName, component.getName()), component.contextPath.fsPath)
+                .then(() => `Service ${serviceName} has been successfully unlinked from the component ${component.getName()}`)
```

Comment:
Wrap compName and comoponent.getName() in ```'```. All ```component``` words should start with capital character ```Component``` in this line and next line.